addons/web_editor/static/src/js/backend.js

```diff
@@ -133,6 +131,8 @@ var FieldTextHtmlSimple = basic_fields.DebouncedField.extend({
         if (history) {
             history.reset();
         }
+        // currently we are unable to use the TranslatableFieldMixin in html widget
```

Comment:
Actually you could, you simply need to override _renderTranslateButton() to render the correct template. Doing so, you don't need to re-defined the handler.